### PR TITLE
Set up a test framework for running meta_net tests

### PR DESCRIPTION
**Pull request by neonphog**
And write a single sanity test that sends a notification between two nodes. Let me know your thoughts on this setup before I go too far creating additional tests.

### CODE REVIEW

File: crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/meta_net/tests.rs [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+            }
+        )*
+    };
+}
+
+type HostRes<T> = Result<T, Box<dyn Send + Sync + std::error::Error>>;
+type HostRet<T> = std::pin::Pin<Box<dyn std::future::Future<Output = HostRes<T>> + 'static + Send>>;
+
+write_test_struct! {
+    KitsuneHost {
```
Comment (ThetaSinner) [inline crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/meta_net/tests.rs:97]: There is a stub implementation for this in `crate::host_api::host_stub`, can that be used here?

File: crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/meta_net/tests.rs [hunk 2]
```diff
@@ -0,0 +1,480 @@
...
+            })
+        }
+        fn lair_tag(&Self,) -> Option<Arc<str>>, Option<Arc<str>> {
+            None
+        }
+        fn lair_client(&Self,) -> Option<lair_keystore_api::LairClient>, Option<lair_keystore_api::LairClient> {
+            None
+        }
+    }
+    InternalHandler {
```
Comment (ThetaSinner) [inline crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/meta_net/tests.rs:200]: For this one there is a stub in `crate::spawn::actor::test_util::internal_stub`, any use here?

**Review by ThetaSinner — COMMENTED** [review]
This looks good. I think this is a sensible level to test the meta net at. There are maybe one or two functions that could be unit tested in the meta net but I think most of the value comes from this type of test 👍🏻

File: crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/meta_net/tests.rs (hunk 1 above)
Comment (neonphog) [reply crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/meta_net/tests.rs:97]: @ThetaSinner - This is set up with default handler callbacks, and the ability to replace them so the test can return whatever it wants or track the input parameters or whatever, more like a mock. So it's more versatile than the stub which must be the same across all tests using it.

File: crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/meta_net/tests.rs (hunk 2 above)
Comment (neonphog) [reply crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/meta_net/tests.rs:200]: Ditto [https://github.com/holochain/holochain/pull/2825#discussion_r1334449678](https://github.com/holochain/holochain/pull/2825#discussion_r1334449678)

File: crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/meta_net/tests.rs (hunk 1 above)
Comment (ThetaSinner) [reply crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/meta_net/tests.rs:97]: Makes sense, I'd like the other mocks/stubs to be similar at some point so maybe we can reconcile them at some point

**Review by ThetaSinner — APPROVED** [review]

**Pull request merged**